test/language/expressions/in/private-field-in-nested.js

```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2021 André Bargull. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+  Can't nest `in` expressions when the left-hand side is PrivateIdentifier.
+info: |
+  Syntax
+    RelationalExpression[In, Yield, Await]:
+    [...]
+    [+In]PrivateIdentifier in ShiftExpression[?Yield, ?Await]
+esid: sec-relational-operators
+negative:
+  phase: parse
+  type: SyntaxError
+features: [class-fields-private, class-fields-private-in]
+---*/
+
+$DONOTEVALUATE();
+
+class C {
+  #field;
+
+  constructor() {
+    #field in #field in this;
```

Comment:
Why would this not evaluate? The RHS is an expression. Or is it that precedence would evaluate the first two pieces first, which would be a syntax error?

Reply:
The [grammar production](https://tc39.es/ecma262/#prod-RelationalExpression) is:
> RelationalExpression :  PrivateIdentifier in ShiftExpression

so it's not possible to nest private field checks.

`a in b in c` is allowed because the left-hand side in this RelationalExpression expression is another RelationalExpression:
> RelationalExpression :  RelationalExpression in ShiftExpression